src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs

```diff
@@ -1,12 +1,1127 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Net.Sockets;
+using System.Reflection;
+
+using Skyline.DataMiner.Net.Parser.ProtocolParsers;
+using Skyline.DataMiner.Net.Topology;
 
 namespace Skyline.DataMiner.Net.Messages
 {
-	public class GetProtocolInfoResponseMessage
+	/// <summary>
+	/// Represents information for a single protocol
+	/// </summary>
+	public class GetProtocolInfoResponseMessage : ResponseMessage
+	{
+		/// <summary>
+		/// Initializes a new instance of the <see cref="GetProtocolInfoResponseMessage"/> class.
+		/// </summary>
+		public GetProtocolInfoResponseMessage() { }
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="GetProtocolInfoResponseMessage"/> class.
+		/// </summary>
+		/// <exception cref="ArgumentNullException"><paramref name="protocolInfo"/> is <see langword="null"/>.</exception>
+		protected GetProtocolInfoResponseMessage(GetProtocolInfoResponseMessage protocolInfo)
+		{
+			if (protocolInfo == null)
+				throw new ArgumentNullException("protocolInfo");
+		}
+
+		/// <summary>
+		/// Links all of the child parameter info object to this object.
```

Comment:
Something seems to be wrong with this sentence. Is this what is meant?
```suggestion
		/// Links all of the child parameter info objects to this object.
```